jiant/tasks/evaluate/core.py

```diff
@@ -406,50 +424,70 @@ def get_labels_from_examples(cls, examples):
     def get_labels_from_cache_and_examples(cls, task, cache, examples):
         return cls.get_labels_from_examples(examples=examples)
 
-    def get_preds_from_accumulator(self, task, accumulator):
-        # TODO: Revisit ReCord scoring  (Issue #51)
-        raise NotImplementedError("Currently need labels ('examples') to compute preds. Refactor.")
+    @classmethod
+    def get_preds_from_accumulator(cls, task, accumulator):
+        logits, entity_strs = accumulator.get_accumulated()
+        guid_list = accumulator.get_guids()
+
+        question_ids = []
+        for guid in guid_list:
+            question_ids.append(guid.split("-")[2])
+
+        # group logits by question id then reorder for submission
+        # need question id, logit, and entity_str
+        # for example, dict of question id to logit and entity_str
+        max_logits = {}
+        for logit, entity_str, question_id in zip(logits, entity_strs, question_ids):
+            if (question_id not in max_logits) or (max_logits[question_id]["logit"][1] < logit[1]):
+                max_logits[question_id] = {"logit": logit, "entity_str": entity_str}
+
+        # Convert labels of max_logits to prediction format
+        preds = []
+        for question_idx, logit_entity in max_logits.items():
+            preds.append({"idx": question_idx, "label": logit_entity["entity_str"]})
+
+        return preds
 
     def compute_metrics_from_accumulator(
-        self, task, accumulator: ConcatenateLogitsAccumulator, tokenizer, labels: list
+        self, task, accumulator: RecordAccumulator, tokenizer, labels: List
     ) -> Metrics:
-        logits = accumulator.get_accumulated()
         predictions_dict, metrics = self.compute_preds_and_metrics_from_logits_and_record_labels(
-            logits=logits, examples=labels,
+            task, accumulator
         )
         return metrics
 
     @classmethod
-    def compute_preds_and_metrics_from_logits_and_record_labels(
-        cls, logits, examples: List[RecordLabelData]
-    ):
-        psg_qns_idx_dict = {}
-        for i, example in examples:
-            psq_qns_idx = example.passage_idx, example.question_idx
-            if psq_qns_idx not in psg_qns_idx_dict:
-                psg_qns_idx_dict[psq_qns_idx] = []
-            psg_qns_idx_dict[psq_qns_idx].append(i)
-
+    def compute_preds_and_metrics_from_logits_and_record_labels(cls, task, accumulator):
```

Comment:
Rename function, or make a separate function, since it's not operating off logits and labels.